ccl/backupccl: TestBackupAsOfSystemTime failed

ccl/backupccl.TestBackupAsOfSystemTime [failed](https://tanzanite.cluster.engflow.com/invocations/default/a61ab389-9a1d-4a83-994a-af35b465b3d6?testReportRun=2&testReportShard=47&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.1 @ [7cf9592bd59077500801e5827e5b8230e9a112f9](https://github.com/cockroachdb/cockroach/commits/7cf9592bd59077500801e5827e5b8230e9a112f9):


```
=== RUN   TestBackupAsOfSystemTime
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupAsOfSystemTime3452600280
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    backup_test.go:3605: retrying txn failed after 50 attempts. original error: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=7010df62 key=/Tenant/10/Table/106/1/1/0 iso=Serializable pri=0.05803686 epo=50 ts=1719130945.952643127,2 min=1719130720.746313387,0 seq=251} lock=true stat=PENDING rts=1719130944.857308166,0 wto=false gul=1719130721.246313387,0 isn=1.
    testutils.go:290: no Invalid Descriptors
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupAsOfSystemTime3452600280
--- FAIL: TestBackupAsOfSystemTime (234.35s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=47</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupAsOfSystemTime.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39760

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -3560,7 +3560,7 @@ func TestBackupAsOfSystemTime(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const numAccounts = 1000
+	const numAccounts = 100
 
 	ctx := context.Background()
 	_, sqlDB, _, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)
@@ -3572,7 +3572,8 @@ func TestBackupAsOfSystemTime(t *testing.T) {
 	sqlDB.QueryRow(t, `SELECT cluster_logical_timestamp()`).Scan(&beforeTs)
 
 	err := crdb.ExecuteTx(ctx, sqlDB.DB.(*gosql.DB), nil /* txopts */, func(tx *gosql.Tx) error {
-		_, err := tx.Exec(`DELETE FROM data.bank WHERE id % 4 = 1`)
+		// Delete a quarter of the accounts
+		_, err := tx.Exec(`DELETE FROM data.bank LIMIT 25`)
 		if err != nil {
 			return err
 		}

